File: app/src/main/java/tech/salroid/filmy/FilmyApplication.java
package tech.salroid.filmy;

import android.app.Application;
import android.content.Context;

import com.crashlytics.android.Crashlytics;

import io.fabric.sdk.android.Fabric;

/**
 * Created by Home on 7/23/2016.
 */
public class FilmyApplication extends Application {

    private static FilmyApplication instance;

    public static FilmyApplication getInstance() {
        return instance;
    }

    public static Context getContext() {
        return instance.getApplicationContext();

    }

    @Override
    public void onCreate() {

        super.onCreate();

        Fabric.with(this, new Crashlytics());

        instance = this;
    }
}

File: app/src/main/java/tech/salroid/filmy/FilmyApplication.kt
package tech.salroid.filmy

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import dagger.hilt.android.HiltAndroidApp
import tech.salroid.filmy.utility.PreferenceHelper

@HiltAndroidApp
class FilmyApplication : Application() {

    override fun onCreate() {
        when (PreferenceHelper.getCurrentThemeMode(this)) {
            getString(R.string.mode_night_no) -> AppCompatDelegate.setDefaultNightMode(
                AppCompatDelegate.MODE_NIGHT_NO
            )
            getString(R.string.mode_night_yes) -> AppCompatDelegate.setDefaultNightMode(
                AppCompatDelegate.MODE_NIGHT_YES
            )
            else -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
        }
        super.onCreate()
    }
}